python-avd/pyavd/_eos_designs/structured_config/base/__init__.py

```
@@ -323,6 +323,62 @@ def ip_name_servers(self) -> None:
+
+        Applies global logging parameters and per-VRF host logging configuration,
+        including source interfaces, protocols, ports, and SSL profiles.
+        Ensures that each VRF has a unique and consistent source interface.
+        """
+        if not self.inputs.logging_settings:
+            return
+
+        settings = self.inputs.logging_settings
+
+        # Apply global logging parameters
+        self.structured_config.logging._update(
+            console=settings.console,
+            monitor=settings.monitor,
+            repeat_messages=settings.repeat_messages,
+            trap=settings.trap,
+            facility=settings.facility,
+            buffered=settings.buffered,
+            synchronous=settings.synchronous,
+            format=settings.format,
+            policy=settings.policy,
+            event=settings.event,
+            level=settings.level,
+        )
+
+        # Temporary structure to group hosts by VRF with source interface tracking
+        vrf_logging_config = EosCliConfigGen.Logging.Vrfs()
+
+        for host in settings.hosts:
+            # Determine the correct VRF and source interface for the host
+            host_vrf, source_interface = self._get_vrf_and_source_interface(
+                vrf_input=host.vrf,
+                vrfs=settings.vrfs,
+                set_source_interfaces=True,
+                context=f"logging_settings.hosts[name={host.name}].vrf",
+            )
+
+            existing_vrf_config = vrf_logging_config.get(host_vrf)
+
+            # Add new VRF config if it doesn't exist, or if source_interface differs (conflict)
+            if not existing_vrf_config or existing_vrf_config.source_interface != source_interface:
+                vrf_logging_config.append_new(name=host_vrf, source_interface=source_interface)
+                self.structured_config.logging.vrfs.append_new(name=host_vrf, source_interface=source_interface)
```

Need clarification:

 We have created `EosCliConfigGen.Logging.Vrfs` object but did not append it in `self. structured_config`, why?

---

Also in line 371 we are appending to `vrf_logging_config` but not using it later, why do we need to append then?

---

To raise the error if conflict in source interface for same vrf we are using the vrf_logging_config. Now suppose if we have both use_mgmt_interface_vrf and use_inband_mgmt_vrf in two different hosts entry and both refer to same vrf like MGMT but the source interface of both vrf can be different hence we need to raise the error for conflicting vrf source interface. You can take a look on negative testcase